internal/worker/dbaccessor/worker.go

```diff
@@ -353,21 +328,20 @@ func (w *dbWorker) loop() (err error) {
 
 			req.done <- nil
 
-		case apiDetails := <-w.apiServerChanges:
-			if err := w.processAPIServerChange(apiDetails); err != nil {
+		case <-firstCheck:
+			firstCheck = nil
+			if err := w.handleClusterConfigChange(false); err != nil {
```

Comment:
If this is requested as a second channel, the fatal argument will not cause it to die,  which seems a bit strange.